src/evt/cam_api.c

```diff
@@ -189,7 +189,81 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
 
 INCLUDE_ASM(s32, "evt/cam_api", func_802CB008, Evt* script, s32 isInitialCall);
 
-INCLUDE_ASM(s32, "evt/cam_api", ShakeCam, Evt* script, s32 isInitialCall);
+ApiStatus ShakeCam(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 camIndex = evt_get_variable(script, *args++);
+    s32 temp_s3 = evt_get_variable(script, *args++);
+    s32 temp_s4 = evt_get_variable(script, *args++);
+    f32 temp_f20 = 2.0f * evt_get_float_variable(script, *args++);
+    Camera* camera = &gCameras[camIndex];
+    f32 temp_f2;
+    f32 phi_f2;
+    s32 temp_a1;
+
+    if (isInitialCall) {
+        switch (temp_s3){
+            case 0:
+            case 1:
+                break;
+
+            case 2:
+                temp_s4 *= 4;
+                break;
+
+            default:
+                break;
+        }
+
+        *(f32*)&script->functionTemp[3] = 1.0f;
```

Comment:
missed float grossness?

Reply:
This one literally has to be like this. I can't get it otherwise.

Reply:
wait so storing it into a float version of the field doesn't produce the same code?

Reply:
Correct, it changes when the save happens. I suspect there is something here with the switch above but I just couldn't get it to work.